node/libs/roles/src/validator/messages/consensus.rs

```diff
@@ -74,96 +75,132 @@ impl Default for Fork {
 
 /// A struct that represents a set of validators. It is used to store the current validator set.
 /// We represent each validator by its validator public key.
-#[derive(Clone, Debug, PartialEq, Eq)]
-pub struct ValidatorSet {
-    vec: Vec<validator::PublicKey>,
-    map: BTreeMap<validator::PublicKey, usize>,
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
+pub struct Committee {
+    vec: Vec<WeightedValidator>,
+    indexes: BTreeMap<validator::PublicKey, usize>,
+    total_weight: u64,
 }
 
-impl ValidatorSet {
-    /// Creates a new ValidatorSet from a list of validator public keys.
-    pub fn new(validators: impl IntoIterator<Item = validator::PublicKey>) -> anyhow::Result<Self> {
-        let mut set = BTreeSet::new();
+impl Committee {
+    /// Creates a new Committee from a list of validator public keys.
+    pub fn new(validators: impl IntoIterator<Item = WeightedValidator>) -> anyhow::Result<Self> {
+        let mut weighted_validators = BTreeMap::new();
+        let mut total_weight: u64 = 0;
         for validator in validators {
```

Comment:
we should also add test checking what happens if some/all validators have weight 0.

Reply:
Should we raise an error on a 0 weight validator?
It seems it shouldn't be allowed, but maybe there is one scenario where it can happen or it is useful (like testing validators without vote or something like that)

Reply:
I added a check (and a test) to forbid 0 weight validators in committee. Let me know if we should do otherwise.

Reply:
sg